libnavigation-ui/src/main/java/com/mapbox/navigation/ui/map/NavigationMapboxMap.java

```diff
@@ -418,6 +418,8 @@ public void restoreFrom(NavigationMapboxMapInstanceState instanceState) {
    * @param route to be drawn
    */
   public void drawRoute(@NonNull DirectionsRoute route) {
+    mapRoute.updateRouteArrowVisibilityTo(true);
+    mapRoute.updateRouteVisibilityTo(true);
```

Comment:
FYI: https://github.com/mapbox/mapbox-navigation-android/pull/3002/files#diff-e5a5b815f804b7277b6f60237aacc739L477 is ready for you 😉

Reply:
> FYI: https://github.com/mapbox/mapbox-navigation-android/blob/master/libnavigation-ui/src/main/java/com/mapbox/navigation/ui/map/NavigationMapboxMap.java#L477 is ready for you 😉

That doesn't solve the issue this PR is trying to address. If you follow the steps to reproduce in #3001 you'll see the route is already drawn on the screen. There's no reason to call showRoute().  The underlying issue is the progress listener gets disabled when the trip session stops and needs to get re-enabled.

Reply:
```java
  public void showRoute() {
    mapRoute.updateRouteVisibilityTo(true);
    mapRoute.updateRouteArrowVisibilityTo(true);
  }
```
this is the implementation of the `showRoute()` method. Is your code different than it if the order matters? 🤔

Reply:
@JunDai This is a good callout. Looks like the any visibility issue _can be solved_ using the interfaces available. The disappearing route line is solvable if you're using NavigationMapboxMap, or NavigationMapRoute. Bug reports may come in from interface confusion.

What might make NavigationMapboxMap confusing, is the routes are visible by default. And then they're automatically turned invisible when the route is removed by free drive. Another fix for the `BasicNavigationActivity` could be to call manually call `showRoute()`

https://github.com/mapbox/mapbox-navigation-android/blob/aacb266f75fce40442f62f5f1a5d3fb2f959a6ca/examples/src/main/java/com/mapbox/navigation/examples/core/BasicNavigationActivity.kt#L147-L150
```kotlin
+navigationMapboxMap?.showRoute()
```

So, should drawRoute() also call showRoute() for you. So you don't need to manually?

It would be good to show this to those reporting the issues.

Reply:
> Another fix for the BasicNavigationActivity could be to call manually call showRoute()

yeap, that's another option which initially I was thinking of and more preferable.
in practice, you call the removeRoute() manually so you are responsible to call the showRoute() too. If so, we need to make sure this is documented.

The other hand, explicitly call showRoute() in `drawRoute()` might also make sense since you are going to drawRoute, I think we need to make sure the route is visible.

I don't have strong opinion one over the other.

Reply:
From discussion, the issue we're trying to resolve here is no longer an issue. The interface is doing what it is designed to do.

the issue is with a visible route, the vanishRouteLineEnabled=true does not work